app/helpers/upload_helper.rb

```diff
@@ -1,18 +1,23 @@
 # Helpers for handling uploading data files for various models.
 module UploadHelper
-  def process_file_upload
+  def process_file_upload(allowed_filetypes = %w[.csv .yml])
     encoding = params[:encoding] || 'UTF-8'
     upload_file = params.require(:upload_file)
 
     if upload_file.size == 0
       raise StandardError, I18n.t('upload_errors.blank')
     end
 
-    filetype = File.extname(upload_file.original_filename)
+    if allowed_filetypes.size == 1
+      filetype = allowed_filetypes[0]
+    else
+      filetype = File.extname(upload_file.original_filename)
+    end
+
     if filetype == '.csv'
       {
         type: '.csv',
-        file: upload_file,
+        contents: upload_file.read,
```

Comment:
This is a good start, but take a look at how the `read` call happens in the YML case. This encoding conversion is very useful for raising an error when given binary data here, rather than when the `contents` are used later.